### DISCUSSION TITLE
Convert a Markdown-formatted manuscript or outline into a novelWriter project

### DISCUSSION THREAD

**Discussion by peter88213** [vkbo/novelWriter#1461] [General]
As a slight modification of my [mindmap](https://github.com/peter88213/mm2nw) and [ODT conversion](https://github.com/peter88213/odt2nw) Python scripts, I quickly threw in a standalone Markdown importer. 

Unlike novelWriter's built-in Markdown import, [md2nw](https://github.com/peter88213/md2nw) creates a project that is completely split into individual documents. In addition, the conversion script handles single-line synopses on all levels.

In any case, it also processes the *.nwd* Markdown documents exported by novelWriter if you change the file extension to *.md*.

**Comment by vkbo** [comment]
novelWriter doesn't really use Markdown any more, or perhaps it can be argued that it is an extension to Markdown. In either case, since Windows has made it increasingly difficult to actually change the file extension of files, I may change the build format to use `.txt` instead.

**Reply by peter88213** [reply]
I do not quite get the point, maybe because on Windows I've been using _Total Commander_ since forever. But that should not be the problem. Actually, my point is to be able to import Markdown-formatted files, usually from other sources than *novelWriter*. Only a small subset of a common Markdown dialect is supported, and this matches the novelWriter format, whatever we want to name it.
Do you think my script should accept various filename extensions for input?

**Reply by vkbo** [reply]
I was mostly talking about the export format from novelWriter. The one format you can select that just assembles the raw novelWriter text files is exported with the `.nwd` extension. It makes more sense to use `.txt`, that's all.

Especially since the installer doesn't register the `.nwd` extensions on any platform, and Windows is terrible at figuring out how to open files on its own.

**Reply by peter88213** [reply]
I see. Yes, basically it's a good idea not to introduce your own file extension for something like this. 

By the way, in reviews of Markdown-based word processors I have even read complaints that the software uses an "unknown proprietary file format *.md*".

**Reply by vkbo** [reply]
Haha, well, far too many reviews are written by people with more opinions than knowledge.

**Reply by peter88213** [reply]
Anyway, now we have three new Python scripts to convert outlines or manuscripts to NovelWriter format. It was a sporty exercise, but I will leave it at that. The three projects on GitHub are self contained, so they have no dependencies on third party libraries. Also, their user interface is set to the command line, so no GUI libraries need to be imported either. 

Anyone who wants to is free to maintain or further develop these tools. They are completely modular, rely on a standardized[ data model](https://github.com/peter88213/PyWriter/tree/main/src/pywriter/model), and the converters expect a [UI facade](https://github.com/peter88213/PyWriter/tree/main/src/pywriter/ui) for messaging, behind which pretty much any Python app can hide.
